optking/displace.py

```diff
@@ -139,6 +133,11 @@ def displace_molsys(oMolsys, dq_in, fq=None, ensure_convergence=False, return_st
 
     # Return final, total displacement ACHIEVED
     dq = q_final - q_orig
+
+    linear_list = linear_bend_check(molsys)
+    if linear_list:
+        raise AlgError("New linear angles", newLinearBends=linear_list)
```

Comment:
Yes. This is tested by the second optimization in `test_opt2_allene.py`. The old check (in StepAlgorithms.py) occurred after displacement of the molecular system but with `dq_achieved` passed in again. In one test using lj gradients, we were adding linear bends unnecessarily because a 30 degree step was being checked as if a 60 degree step had been taken. I moved it into displace because it seemed more natural.